### ISSUE TITLE
[Bug]: allowRendererProcessReuse changes CORS behavior for file:// URLs on second navigation

### ISSUE THREAD

**Issue by matheusd** [electron/electron#28641]
### Preflight Checklist

I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Electron Version

12.0.2

### What operating system are you using?

Other Linux

### Operating System Version

Fedora 32

### What arch are you using?

x64

### Expected Behavior

Access to iframe contents should have a consistent CORS behavior for `file://` URLs irrespective of the value of of the `app.allowRendererProcessReuse` setting.

CORS behavior for `file://` URLs is implementation dependent. I'm aware there are different approaches to handling this, and could _a priori_ accept any one as long as it's consistently enforced.

Personally, I believe that for electron apps each `file://` page should be considered an individual origin with respect to CORS restrictions. Thus, iframes loaded from a `file://` URL would be barred from directly accessing contents of the parent document (and vice-versa).

### Actual Behavior

Behavior for CORS enforcement of iframe content access depends on the setting of `app.allowRendererProcessReuse`.

If `app.allowRendererProcessReuse = true`, then access between child and parent documents is always allowed.

On the other hand, if `app.allowRendererProcessReuse = false`, then on the first access upon app start (i.e. on the execution initiated by `mainWindow.loadURL()`) the documents are allowed access between each other but crucially, after a page reload (that is, once some navigation event happens and presumably a new renderer process is started) the documents are **not** allowed to access each other.

I've included a gist link, and here's a repo with a readme and images of what I'm seeing on my sample.

https://github.com/matheusd/electron-cors-loadurl-bug

### Testcase Gist URL

https://gist.github.com/matheusd/3e9634d0ff0b16ce74da33497cb63cfe

**Comment by Prinzhorn** [comment]
I _think_ you mean the SOP (same-origin policy) not CORS (cross-origin resource sharing)? CORS is only relevant for ajax/fetch.

> Personally, I believe that for electron apps each `file://` page should be considered an individual origin with respect to CORS restrictions. Thus, iframes loaded from a `file://` URL would be barred from directly accessing contents of the parent document (and vice-versa).

You are disabling the SOP by setting `webSecurity: false` (which I think also makes CORS irrelevant for that matter). So the expected behavior is what you see with `app.allowRendererProcessReuse = true`. Every origin can access any other. E.g. Google can read Facebook cookies or an evil website can read all your files via `file://` ajax/iframe etc.

Interestingly you shouldn't be able to change `allowRendererProcessReuse` at all in Electron 12 according to the timeline here https://github.com/electron/electron/issues/18397#issue-446875840. But this might be outdated (I think it shifted one version, so maybe it'll be that way in v13).

So to summarize I think this is a bug with `allowRendererProcessReuse` which will be removed anyway.

**Comment by MarshallOfSound** [comment]
@Prinzhorn is correct.  The behavior with `allowRendererProcessReuse = true` is the correct and expected behavior, the `false` behavior is a side effect of the way that flag works, is a bug, and is a sympton of the underlying reasons why we introduced the `allowRendererProcessReuse` migration in the first place.  See #18397 for more info

**Comment by matheusd** [comment]
> You are disabling the SOP by setting webSecurity: false

This also happened with `webSecurity: true`. 

> Is there a reason you disabled every security feature?

We never access or display external resources from our electron app (only packaged resources), and are in the middle of migrating the code so that we can in fact run with `nodeIntegration: false` and `contextIsolation: true` in the main UI.

>  The behavior with allowRendererProcessReuse = true is the correct and expected behavior,

Noted. Thank you for the prompt response!
